ClassTranscribeDatabase/Models/FileRecord.cs

```diff
@@ -42,6 +44,35 @@ public static async Task<FileRecord> GetNewFileRecordAsync(string filepath, stri
             return fileRecord;
         }
 
+        public static async Task SetFilePath(CTDbContext context, Entity entity)
+        {
+            // If the entity's CreatedAt field is set to the default DateTime (Jan 1, 0001) this means that
+            // it is a new Entity that hasn't yet been added to the DB so we can just use the current date
+            var createdAt = entity.CreatedAt != default ? entity.CreatedAt : DateTime.Now;
+            var filePath = $"{createdAt:yyMM}-{CommonUtils.RandomString(4)}";
+            var newDirectory = Globals.appSettings.DATA_DIRECTORY + System.IO.Path.DirectorySeparatorChar;
+
+            switch (entity)
+            {
+                case Course course:
+                    course.FilePath = filePath;
+                    newDirectory += course.FilePath;
+                    break;
+                case CourseOffering courseOffering:
+                    var linkedCourse = await context.Courses.FindAsync(courseOffering.CourseId);
+                    Contract.Requires(linkedCourse != null);
+
+                    courseOffering.FilePath = System.IO.Path.Combine(linkedCourse.FilePath, filePath);
+                    newDirectory += courseOffering.FilePath;
+                    break;
+                default:
+                    throw new InvalidOperationException("Invalid entity passed: " + entity.GetType());
+            }
+
+            Directory.CreateDirectory(newDirectory);
```

Comment:
Does this silently fail if the directory already exists? If so that should be documented.
If this is not desired behavior then it could repeat with a new random path.

Reply:
Updated this method so that new random paths are tried until finding a directory path that is unique